src/features/comments/commentsSlice.ts

```diff
@@ -0,0 +1,99 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import {
+  createComment, deleteComment, getPostComments,
+} from '../../api/comments';
+
+type Comments = {
+  comments: Comment[],
+  isLoading: boolean,
+  hasError: boolean,
+};
+
+const initialState: Comments = {
+  comments: [],
+  isLoading: false,
+  hasError: false,
+};
+
+export const fetchComments = createAsyncThunk(
+  'comments/fetch', (id: number) => {
+    return getPostComments(id);
+  },
+);
+
+export const addComment = createAsyncThunk(
+  'comments/add', (data: Omit<Comment, 'id'>) => {
+    return createComment(data);
+  },
+);
+
+export const delComment = createAsyncThunk(
+  'comments/delete', (id: number) => {
+    return deleteComment(id);
+  },
+);
+
+const postSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    removeComment: (state, action) => {
```

Comment:
you don't need it - use `deleteComment` as an action when you want to delete a comment